bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/exception/HiveClientException.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client.exception;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ *
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public abstract class HiveClientException extends RuntimeException {
```

Comment:
Is there a reason that this is a runtime exception rather than a checked exception? Runtime exceptions are generally treated as unrecoverable, is that really the case for a communication error during an api call? I think it should be a checked exception instead.